snapcraft/internal/errors.py

```diff
@@ -41,13 +41,24 @@ def __init__(self, **kwargs):
     def __str__(self):
         return self.fmt.format([], **self.__dict__)
 
+    @property
```

Comment:
this doesn't necessarily need to be a property, it might as well be a class attribute, yes you can change it, but someone can override this behavior too.

I am less inclined about properties these days.